prime-router/src/main/kotlin/ReportStreamFilter.kt

```diff
@@ -0,0 +1,97 @@
+ * (and someday, per Sender too?)
+ *
+ * Examples of FilterTypes that can be applied:
+ *  @param jurisdictionalFilter - used to limit the data received or sent by geographical region
+ *  @param qualityFilter - used to limit the data received or sent, by quality
+ *  @param routingFilter - used to limit the data received or sent, by who sent it.
+ * We allow a different set of filters per [topic]
+ */
+data class ReportStreamFilters(
+    val topic: String,
+    val jurisdictionalFilter: ReportStreamFilter?,
+    val qualityFilter: ReportStreamFilter?,
+    val routingFilter: ReportStreamFilter?,
+) {
+
+    companion object {
+
+        // For each 'topic' that ReportStream handles, there is a set of default filters.
+        // Currently these are defined here in code.
+        // Each Organization and Receiver can override the defaults.
+        // todo move all of these to a GLOBAL Setting in the settings table
+        val defaultCovid19QualityFilter: ReportStreamFilter = listOf(
+            // valid basic test info
+            "hasValidDataFor(message_id)",
+            "hasValidDataFor(equipment_model_name)",
+            "hasValidDataFor(specimen_type)",
+            "hasValidDataFor(test_result)",
+            // valid basic human info
+            "hasValidDataFor(patient_last_name, patient_first_name)",
+            "hasValidDataFor(patient_dob)",
+            // has minimal valid location or other contact info (for contact tracing)
+            "hasAtLeastOneOf(patient_street,patient_zip_code,patient_phone_number,patient_email)",
+            // has valid date (for relevance/urgency)
+            "hasAtLeastOneOf(order_test_date,specimen_collection_date_time,test_result_date)",
+            // has at least one valid CLIA
+            "isValidCLIA(testing_lab_clia,reporting_facility_clia)",
+            // never send T (Training/Test) or D (Debug) data to the states.
+            "doesNotMatch(processing_mode_code,T,D)",
+        )
+        private val defaultCovid19Filters = ReportStreamFilters(
+            topic = "covid-19",
+            jurisdictionalFilter = listOf("allowNone()"), // Receiver *must* override this to get data!
+            qualityFilter = defaultCovid19QualityFilter,
+            routingFilter = listOf("doesNotMatch(processing_mode_code, T, D)"), // Default is no Training/Debug data
+        )
+
+        private val defaultCsvFileTestFilters = ReportStreamFilters(
+            topic = "CsvFileTests-topic",
+            jurisdictionalFilter = listOf("allowAll()"),
+            qualityFilter = listOf("hasValidDataFor(lab,state,test_time,specimen_id,observation)"),
+            routingFilter = listOf("allowAll()")
+        )
+
+        private val defaultTestFilters = ReportStreamFilters(
+            topic = "test",
+            jurisdictionalFilter = null,
+            qualityFilter = listOf("matches(a, no)"),
+            routingFilter = listOf("matches(b, false)")
+        )
```

Comment:
You'll see three sets of filters here:  One "real" one for covid-19, then a couple test filter sets.

Note that each ReportStreamFilters obj has to be associated with a 'topic', because filters surely vary by topic.

All of this belongs in Settings, not here - a known ticketed issue.